ci/release/update-version.sh

```diff
@@ -6,7 +6,53 @@
 ################################################################################
 
 ## Usage
-# bash update-version.sh <new_version>
+# Primary interface: ./ci/release/update-version.sh --run-context=main|release <new_version>
+# Fallback: Environment variable support for automation needs
+# NOTE: Must be run from the root of the repository
```

Comment:
question: Can we check where we are when we are run and error if not from the correct directory?